src/containers/Dashboard/NodeSlice.ts

```diff
@@ -35,41 +35,51 @@ const nodeSlice = createSlice({
     },
 });
 
-export const {nodeListComplete, storeNodeList, storeNodeInfo, storeDiskInfo} = nodeSlice.actions;
+export const {isLoading, storeNodeList, storeNodeInfo, storeDiskInfo} = nodeSlice.actions;
 export default nodeSlice.reducer;
 
-export const getNodeList = (auth: string | null): AppThunk => async dispatch => {
+export const getNodeList = (): AppThunk => async (dispatch, getState) => {
     try {
-        const nodeListResponse = await getNodes(auth);
+        const token = getState().user.token;
+        const nodeListResponse = await getNodes(token);
         dispatch(storeNodeList(nodeListResponse.data));
+        dispatch(isLoading());
+    } catch (err) {
+        throw err;
+    }
+};
 
-        if (nodeListResponse.data[0] && nodeListResponse.data[0].id) {
-            const nodeInfoResponse = await getMinerInfo(auth, nodeListResponse.data[0].id);
-            dispatch(
-                storeNodeInfo({
-                    version: nodeInfoResponse.data.version,
-                    sectorSize: nodeInfoResponse.data.sectorSize,
-                    minerPower: nodeInfoResponse.data.minerPower,
-                    totalPower: nodeInfoResponse.data.totalPower,
-                }),
-            );
-
-            const diskDetailsList: Array<INodeDiskState> = [];
-            const nodeList = nodeListResponse.data;
-
-            for (let index = 0; index < nodeList.length; index++) {
-                const response: any = await getDiskDetails(auth, nodeList[index].id);
-                diskDetailsList.push({
-                    id: response.data[0].id,
-                    freeSpace: response.data[0].freeSpace,
-                    takenSpace: response.data[0].takenSpace,
-                    nodeId: response.data[0].nodeId,
-                });
-            }
+export const getGeneralInfo = (nodeId: number): AppThunk => async (dispatch, getState) => {
+    try {
+        const token = getState().user.token;
+        const nodeInfoResponse = await getMinerInfo(token, nodeId);
+        dispatch(
+            storeNodeInfo({
+                version: nodeInfoResponse.data.version,
+                sectorSize: nodeInfoResponse.data.sectorSize,
+                minerPower: nodeInfoResponse.data.minerPower,
+                totalPower: nodeInfoResponse.data.totalPower,
+            }),
+        );
+    } catch (err) {
+        throw err;
+    }
+};
 
-            dispatch(storeDiskInfo(diskDetailsList));
-            dispatch(nodeListComplete());
+export const getDiskInfo = (nodeList: Array<INodeState>): AppThunk => async (dispatch, getState) => {
+    try {
+        const token = getState().user.token;
+        const diskDetailsList: Array<INodeDiskState> = [];
+        for (let index = 0; index < nodeList.length; index++) {
+            const response: any = await getDiskDetails(token, nodeList[index].id);
+            diskDetailsList.push({
```

Comment:
You can push all information here that you get. We will used `updatedAt` field as well at some point to display to user how much information is old